lib/state_machines/machine.rb

```diff
@@ -2046,14 +2046,12 @@ def action_hook
     # the method and is further along in the ancestor chain than this
     # machine's helper module.
     def owner_class_ancestor_has_method?(scope, method)
+      return false unless owner_class_has_method?(scope, method)
```

Comment:
The idea here is that if the owner class doesn't respond to this methods, there is no point walking down the ancestor list to figure out which ancestors defines it.